daemon/graphdriver/zfs/zfs.go

```diff
@@ -361,6 +362,15 @@ func (d *Driver) Remove(id string) error {
 	name := d.zfsPath(id)
 	dataset := zfs.Dataset{Name: name}
 	err := dataset.Destroy(zfs.DestroyRecursive)
+	if err != nil {
+		var errZfs *zfs.Error
+		isZfsError := errors.As(err, &errZfs)
+		if isZfsError && strings.HasSuffix(errZfs.Stderr, "dataset does not exist\n") {
+			logger := log.G(context.TODO()).WithField("storage-driver", "zfs")
+			logger.WithError(err).Warnf("Tried to destroy inexistant dataset %q", name)
```

Comment:
Actually; let me change this to `WithFields` instead of chaining `WithField` and `WithError`